astropy/io/fits/hdu/hdulist.py

```diff
@@ -986,20 +984,18 @@ def _flush_resize(self):
             else:
                 new_file = name
 
-            hdulist = self.fromfile(new_file, mode='append')
-
-            for hdu in self:
-                hdu._writeto(hdulist._file, inplace=True, copy=True)
-
-            if sys.platform.startswith('win'):
-                # Collect a list of open mmaps to the data; this well be used
-                # later.  See below.
-                mmaps = [(idx, _get_array_mmap(hdu.data), hdu.data)
-                         for idx, hdu in enumerate(self) if hdu._has_data]
-
-            hdulist._file.close()
-            self._file.close()
+            with self.fromfile(new_file, mode='append') as hdulist:
```

Comment:
While I haven't found an issue about this, there was a small bug here. This method left a few open files behind in the case of having an exception in the loop in the line below. Adding the with statement forced those temporary files to be closed.